### DISCUSSION TITLE
Consideration for JsonPathKt

### DISCUSSION THREAD

**Discussion by djrmun** [nomemory/mapneat#15] [Ideas]
I've recently begun to experiment with MapNeat and I'm finding the use of operator overloading and nested assignments quite helpful as they allow me to mirror (in code) the JSON structure I'm producing.  Kudos!  I'm considering using MapNeat for payload transformations when handling server-side termination of a REST endpoint where the handling requires the issuance of a client (REST) transaction to another endpoint based on a (possibly MapNeat) transformation of the payload.

I'm concerned about TPS and overall throughput and I'm wondering if you have performed/published any benchmarks on MapNeat.  Since I realize that one of the main (time) contributors is likely JsonPath, I am wondering if you have seen or considered [codeniko/JsonPathKt](https://github.com/codeniko/JsonPathKt) as it purports to be "A lighter and more efficient implementation of JsonPath in Kotlin."

**Comment by nomemory** [comment]
No to be honest I didn't considered the library you've mentioned. But it's a good idea to be allow users to switch between the underlying JsonPath implementations. Or maybe switch all together to JsonPathKt. At some point I've even contemplated on writing an implementation myself, but afaik there's no "official" target for ANTRL. 

Anyway, it's a good suggestion and I will definitely allocate some time to this.

Now, regarding the the benchmarks, is there anything you want me to focus on?

Ps: thank you for taking time to look into this.
